x-pack/plugins/ml/server/models/model_management/models_provider.ts

```diff
@@ -51,5 +50,17 @@ export function modelsProvider(client: IScopedClusterClient) {
 
       return modelIdsMap;
     },
+
+    /**
+     * Deletes associated pipelines of the requested model
+     * @param modelIds
+     */
+    async deleteModelPipelines(modelIds: string[]) {
+      const pipelines = await this.getModelsPipelines(modelIds);
+      const pipelinesIds: string[] = [...pipelines.values()].flatMap((v) => Object.keys(v!));
```

Comment:
is there a risk of duplicate pipeline ids here?
If a pipeline uses two models and both are in the list for deletion, this list might contain the same pipeline ID twice.
This would then throw an error trying to delete the second instance of that ID.

Reply:
fixed in https://github.com/elastic/kibana/pull/158671/commits/d4e8179f4753502d4d44f84f977d58c603dd4bf0